lib/assessment/label.py

```diff
@@ -358,20 +381,35 @@ def compute_messages(self, prompt, rubric, student_code, examples=[]):
         return messages
 
     def get_response_data_if_valid(self, choice, rubric, student_id, choice_index=None, reraise=False, response_type='tsv'):
-        response_text = choice['message']['content']
+        response_text = None
+        finish_reason = None
+        if 'message' in choice.keys(): # OpenAI response
+            response_text = choice['message']['content']
+            finish_reason = choice['finish_reason']
+        elif 'completion' in choice.keys(): # Claude 2 response
+            response_text = choice['completion']
+            finish_reason = choice['stop_reason']
+        elif 'content' in choice.keys(): # Claude 3 response
+            response_text = choice['content'][0]['text']
+            finish_reason = choice['stop_reason']
+
         try:
             choice_text = f"Choice {choice_index}: " if choice_index is not None else ''
             if not response_text:
                 raise InvalidResponseError("empty response")
             text = response_text.strip()
 
             if response_type == 'json':
-                response_data = self.parse_json_response(text, student_id, finish_reason=choice['finish_reason'])
+                response_data = self.parse_json_response(text, student_id, finish_reason=finish_reason)
             elif response_type == 'tsv':
                 response_data = self.parse_non_json_response(text)
             else:
                 raise ValueError(f"Invalid response type: {response_type}")
 
+            if type(response_data) == 'list' and "Key Concept" in response_data[0]:
+                for i in range(len(response_data)):
+                    response_data[i]["Key Concept"] = response_data[i]["Key Concept"].replace("\"", "“", 1).replace("\"", "”", 1)
+
```

Comment:
Why replace the straight quotes with smart quotes?

Reply:
Claude returns canceled straight quotes in Key Concepts that have quotes, which causes a mismatch, because we use smart quotes. I tried changing it the other ways (swap out our smart quotes with straight quotes) but this was a quicker fix, and it doesn't cause issues with other models.

Reply:
That makes sense. Imo stripping out all quotes and whitespace from the key concepts/ scores would be better, I guess I'm not sure why there are quotes in those strings at all? But also, 🤷‍♀️ a quick fix here seems fine

Reply:
> Claude returns canceled straight quotes in Key Concepts that have quotes

@Nokondi can you please give an example showing what you mean by this?

> That makes sense. Imo stripping out all quotes and whitespace from the key concepts/ scores would be better, I guess I'm not sure why there are quotes in those strings at all? But also, 🤷‍♀️ a quick fix here seems fine

@cearachew I agree here also. please correct my understanding but this would mean modifying the system prompt until the AI model(s) no longer returned key concepts in quotes, is that correct? perhaps this is something we could put on a list to do "before we expand our pool of lessons".

Reply:
I don't think this means modifying the prompts at all. It just means post-processing to normalize the model outputs and the comparison strings from the rubrics to strip surrounding whitespace/quotes/whatever

Reply:
ok, yes, I see what you're saying and think it sounds doable without waiting for lesson expansion, though also agree it doesn't seem like a blocker for this PR

Reply:
For whatever reason, our rubrics and expected grades files use smart quotes in some key concepts, so if the LLM response doesn't return identical key concepts, you end up with a error. Doesn't seem to have been an issue with GPT, but Claude only returns straight quotes. I'm fine with removing the quotes as an alternate solution.

Reply:
I see now we are talking about quotes within some Key Concept names, rather than surrounding them:
* https://github.com/code-dot-org/aitt_release_data/blob/758ea5a226bada8f457e14894d1750b60c225cf2/csd3-2023-L14/standard_rubric.csv#L4
* 
```
(.venv) Dave-MBP:~/src/aiproxy/datasets/contractor-grades-batch-1-fall-2023 (main)$ cat csd3-2023-L14/actual_labels.csv
student,Modularity - Sprites and Sprite Properties,Position and Movement,Optional “Stretch” Feature - Variables
...
```
normalizing to curly quotes in this way seems a bit fragile because it assumes there is exactly 0 or 1 pair of curly quotes within the string and that there are no straight quotes. OTOH if we normalize to anything else (straight quotes or no quotes) it will be slightly more work because we'll also have to normalize what we read out of the standard_rubric.csv and actual_labels.csv files. 

since we'll get pretty immediate validation error if we introduce straight quotes into a key concept in the future, and this prob won't happen until we expand to more lessons, I think I feel good about shipping what you have here Mark, and opening a work item to instead normalize to straight quotes as part of our "before we expand to more lessons" list of work items. 

how does this sound to you all?

Reply:
I think that sounds good, I agree that normalizing to replace all quotes with straight quotes for comparison is less fragile than the other way around

Reply:
Sounds fine to me. Yeah, I realized it's a hacky solution, but felt safe for now, since we only have one pair of quotes per Key Concept. I think the easiest fix down the road is to just remove punctuation from the Key Concepts in the rubric and labels files.